core/src/main/java/google/registry/persistence/EntityManagerFactoryProvider.java

```diff
@@ -70,4 +85,57 @@ public static EntityManagerFactory create(String jdbcUrl, String username, Strin
         "Persistence.createEntityManagerFactory() returns a null EntityManagerFactory");
     return emf;
   }
+
+  /** Constructs the {@link EntityManagerFactory} instance for the App Engine application. */
+  public static EntityManagerFactory createForAppEngine() {
```

Comment:
What are you envisioning would call this function?  Does this function have to be static?  If you could change it to an instance method then you could use Dagger to @Inject the various configuration options it depends on.